File: old_trunk/cost/iona/orbacus/java/poa/TestLocationForwardActivator_impl.java
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;

final class TestLocationForwardActivator_impl
    extends org.omg.PortableServer.ServantActivatorPOA
{
    private boolean activate_;

    private Servant servant_;

    private org.omg.CORBA.Object forward_;

    TestLocationForwardActivator_impl()
    {
        activate_ = false;
    }

    public void
    setActivatedServant(Servant servant)
    {
	servant_ = servant;
    }

    public void
    setForwardRequest(org.omg.CORBA.Object forward)
    {
	forward_ = forward;
    }

    public Servant
    incarnate(byte[] oid, POA poa)
	throws ForwardRequest
    {
	activate_ = !activate_;
	if(!activate_)
	    throw new ForwardRequest(forward_);
	return servant_;
    }

    public void
    etherealize(byte[] oid, POA poa, Servant servant, boolean cleanup,
		boolean remaining)
    {
    }
}

File: cost/tests/orbacus/poa/TestLocationForwardActivator_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestLocationForwardActivator_skel.h>
#include <TestLocationForwardActivator_impl.h>

using namespace CORBA;
using namespace PortableServer;

TestLocationForwardActivator_impl::TestLocationForwardActivator_impl()
    : activate_(false)
{
}

void
TestLocationForwardActivator_impl::setActivatedServant(ServantBase* servant)
{
    servant -> _add_ref();
    servant_ = servant;
}


Servant
TestLocationForwardActivator_impl::incarnate(const ObjectId& /*oid*/,
                                             POA_ptr /*poa*/)
{
    activate_ = !activate_;
    if(!activate_)
        throw ForwardRequest(forward_);
    return servant_;
}

void
TestLocationForwardActivator_impl::etherealize(const ObjectId& /*oid*/,
                                               POA_ptr /*poa*/,
                                               Servant /*servant*/,
                                               Boolean /*cleanup*/,
                                               Boolean /*remaining*/)
{
}

void
TestLocationForwardActivator_impl::setForwardRequest(Object_ptr forward)
{
    forward_ = Object::_duplicate(forward);
}
